### PR TITLE
Extra message queue envelopes (C4-27)

### PR DESCRIPTION
**Pull request by netsettler**
This changes the details of how `/dlq_to_primary` works, having is strip the header/envelope information from the message before putting it into the new queue, so that failing messages don't end up accumulating ever more wrappers every time they're requeued, probably making them take a long time and have no effect.

Naturally this involved changes to the test case as well, but I didn't like the double `json.loads` that was happening before. This code has been replaced.

```
body = json.loads(json.loads(msg['Body'])['Body'])['uuid']
```

(Among other thing, I think on every trip through the DLQ an extra wrapper was added so the above code snippet could only have worked on one of those iterations.)

There are still some other issues we should follow up separately, relate to why send_message takes a JSON body but receive_messages expects to receive a body as a string that needs to be unpacked. I think receive_messages should do the unpacking itself, but that would require a careful search of all callers and I'm not up for that tonight, so I put in a `TODO` comment instead.

I've tentatively disabled the `flaky` marker on this test. It feels like something that should just work. I guess we'll see. But it passes tests now (in the middle of the night, if that matters).

### CODE REVIEW

**Comment by netsettler** [comment]
And I bumped pytz to 2020.1 at no extra cost. :) There's really no reason not to be up-to-date on that, and both Fourfront and CGAP have been selecting that version anyway. More things use dcicutils, so I'll have to do more research to see if I can bump it there, but I'll take this small win for now.

File: snovault/elasticsearch/indexer_queue.py [hunk 1]
```diff
@@ -134,8 +134,40 @@ def dlq_to_primary(context, request):
     Endpoint to move all uuids on the DLQ to the primary queue
     """
     queue_indexer = request.registry[INDEXER_QUEUE]
-    dlq_messages = queue_indexer.receive_messages(target_queue='dlq')
+    # What comes out of .receive_messages() looks like:
```
Comment (willronchetti) [inline snovault/elasticsearch/indexer_queue.py:137]: This comment should go in the docstring

File: snovault/tests/test_indexing.py [hunk 2]
```diff
@@ -336,28 +336,45 @@ def test_dlq_to_primary(app, anontestapp, indexer_testapp):
     """
     indexer_queue = app.registry[INDEXER_QUEUE]
     indexer_queue.clear_queue()
-    test_bodies = ["destined for primary!", "i am also destined!"]
-    success, failed = indexer_queue.add_uuids(app.registry, test_bodies,
-                                        target_queue='dlq')
-    assert not failed
-    assert len(success) == 2
+    test_uuids = ["destined for primary!", "i am also destined!"]
+    print("Setup phase. Placing 2 dummy UUIDs directly into the DLQ...")
```
Comment (willronchetti) [inline snovault/tests/test_indexing.py:340]: These prints should all be removed

File: snovault/elasticsearch/indexer_queue.py (hunk 1 above)
Comment (netsettler) [reply snovault/elasticsearch/indexer_queue.py:137]: Personally, I think not. The docstring should contain information useful to the consumer of the function, who does not have access to or interest in the implementation. You'll often see me split information side-by-side with some info in a doc string and some in a comment, and I intend the comment part for the person who has the editor open to the implementation and needs the internal details. A key reason for functionality of this kind  to exist is to hide such detail. I checked when I wrote this that all this information seemed just implementation notes and not anything a person calling this function should need, but armed with this explanation, see if you agree.

**Pull request merged**